accounting/on_chain.go

```diff
@@ -0,0 +1,42 @@
+package accounting
+
+import (
+	"time"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+)
+
+// filterOnChain filters a set of on chain transactions to get only those
+// which lie within [startTime, endTime). Unconfirmed transactions are also
+// excluded from this set.
+func filterOnChain(startTime, endTime time.Time,
+	txns []*lnrpc.Transaction) []*lnrpc.Transaction {
+
+	var filtered []*lnrpc.Transaction
+
+	for _, tx := range txns {
+		timestamp := time.Unix(tx.TimeStamp, 0)
+
+		// Unconfirmed transactions are listed with 0 confirmations,
+		// they have no timestamp so we skip them.
+		if tx.NumConfirmations == 0 {
+			continue
+		}
+
+		// Our start time is inclusive, skip any transactions that are
+		// strictly before our start time.
+		if timestamp.Before(startTime) {
+			continue
+		}
+
+		// Our end time is exclusive, so we skip any transactions that
+		// are after or equal to our end time.
+		if timestamp.After(endTime) || timestamp.Equal(endTime) {
```

Comment:
Could just write: `!timestamp.Before(endTime)`

Reply:
I knew there was a better way of doing this, brain was clearly on powersaver mode :coffee: